### PR TITLE
feat: constraint endpoint

### PR DESCRIPTION
**Pull request by SuperMarcomen**
…nstraints/{signatureId}/instances

### CODE REVIEW

File: server/src/main/java/de/uftos/services/ConstraintInstanceService.java [hunk 1]
```diff
@@ -123,16 +128,21 @@ public ConstraintInstance create(String signatureId, ConstraintInstanceRequestDt
...
+    //  GOOD LUCK!
     Specification<ConstraintInstance> specification = new SpecificationBuilder<ConstraintInstance>()
-        .andJoinEquals(signatureId, "signature", "name")
-        .optionalAndJoinLikeIgnoreCase(argument, "arguments", "value")
         .build();
 
-    return this.repository.findAll(specification, pageable);
+    Page<ConstraintInstance> constraintInstances = this.repository.findAll(specification, pageable);
+    List<ConstraintArgumentDisplayName> displayNames =
+        processConstraintInstances(constraintInstances.getContent());
```
Comment (danieldietzler) [inline server/src/main/java/de/uftos/services/ConstraintInstanceService.java:144]: `process` is like extremely generic. Could we be more specific in the naming here?

File: server/src/main/java/de/uftos/services/ConstraintInstanceService.java [hunk 2]
```diff
@@ -177,11 +186,67 @@ public ConstraintInstance update(String signatureId, String id,
...
+        Teacher teacher = teacherRepository.findById(id).orElseThrow();
+        yield new ConstraintArgumentDisplayName(id,
+            "%s %s".formatted(teacher.getFirstName(), teacher.getLastName()));
+      }
+      case STUDENT_GROUP -> new ConstraintArgumentDisplayName(id,
+          studentGroupRepository.findById(id).orElseThrow().getName());
+      case TIMESLOT -> new ConstraintArgumentDisplayName(id,
+          timeslotRepository.findById(id).orElseThrow().getDay().toString());
+    };
+  }
```
Comment (danieldietzler) [inline server/src/main/java/de/uftos/services/ConstraintInstanceService.java:249]: Gotta love those lol

File: server/src/main/java/de/uftos/services/ConstraintInstanceService.java [hunk 3]
```diff
@@ -177,11 +186,67 @@ public ConstraintInstance update(String signatureId, String id,
...
+    return displayNames;
+  }
+
+  private ConstraintArgumentDisplayName getDisplayName(ConstraintArgument constraintArgument) {
+    String id = constraintArgument.getValue();
+    return switch (constraintArgument.getConstraintParameter().getParameterType()) {
+      case TAG ->
+          new ConstraintArgumentDisplayName(id, tagRepository.findById(id).orElseThrow().getName());
+      case ROOM -> new ConstraintArgumentDisplayName(id,
+          roomRepository.findById(id).orElseThrow().getName());
```
Comment (danieldietzler) [inline server/src/main/java/de/uftos/services/ConstraintInstanceService.java:227]: We shouldn't have empty `orElseThrow` calls. Either remove them, use `.get()` and accept the runtime error (which I'm not sure if that's fine or not), or throw a bad request.

File: server/src/main/java/de/uftos/services/ConstraintInstanceService.java (hunk 1 above)
Comment (SuperMarcomen) [reply server/src/main/java/de/uftos/services/ConstraintInstanceService.java:144]: done

File: server/src/main/java/de/uftos/services/ConstraintInstanceService.java (hunk 3 above)
Comment (SuperMarcomen) [reply server/src/main/java/de/uftos/services/ConstraintInstanceService.java:227]: Now throwing a bad request

File: server/src/main/java/de/uftos/services/ConstraintInstanceService.java (hunk 2 above)
Comment (SuperMarcomen) [reply server/src/main/java/de/uftos/services/ConstraintInstanceService.java:249]: I tried to make the switch as tidy and simple as possible. I don't know if it's possible to improve it

Comment (danieldietzler) [reply server/src/main/java/de/uftos/services/ConstraintInstanceService.java]: Yeah I don't think so either tbh

**Review by danieldietzler — APPROVED** [review]
Ty!

**Pull request merged**